central/imageintegration/datastore/singleton.go

```diff
@@ -22,16 +23,40 @@ var (
 	dataStore DataStore
 )
 
-func initializeDefaultIntegrations(storage store.Store) {
+func initializeIntegrations(storage store.Store) {
 	ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowAllAccessScopeChecker())
 	iis, err := storage.GetAll(ctx)
 	utils.CrashOnError(err)
+	// If we are starting from scratch in online-mode, add the default image integrations.
 	if !env.OfflineModeEnv.BooleanSetting() && len(iis) == 0 {
 		// Add default integrations
 		for _, ii := range store.DefaultImageIntegrations {
 			utils.Must(storage.Upsert(ctx, ii))
 		}
 	}
+
+	// If the feature flag is disabled, remove all "sourced" autogenerated registries from the datastore.
+	if !features.SourcedAutogeneratedIntegrations.Enabled() {
+		if len(iis) > 0 {
+			log.Infof("[STARTUP] Starting deletion of 'sourced' image integrations")
+		}
+
+		var attempted, deleted int
+		for _, ii := range iis {
+			if ii.GetAutogenerated() && ii.GetSource() != nil {
```

Comment:
When there's customer who migrate from 3.72.0 -> 3.73.0, they will have the mix of autogenerated integrations with `Source != nil` and `Source == nil`.

In case of a "clean" installation of 3.73.0, _all_ autogenerated image integrations will have `Source != nil` set.

So, we will delete _all_ autogenerated image integrations then.

In that case, how will the recovery look like for this? Will it be eventual consistent such that the autogenerated registries we expect to have (i.e. for internal OpenShift registry) will show up after some time, or does that require bumps of sensors to send the autogenerated registries once more?

(I honestly don't think that the OpenShift internal ones are _that_ important from an autogenerated image integration perspective, unless central can reach them, which won't be the case for most of them)

Reply:
In the case of a clean install, I'd expect us to eventually find all the autogenerated registries. When it comes to OpenShift internal registry, we do not use the stuff sent to Central for that